shm/src/store.cpp

```diff
@@ -88,12 +98,72 @@ void Store::nextDay() {
     }
 }
 
+void Store::generateRegularCargo() {
+    std::vector<std::shared_ptr<Cargo>> possibleRegularCargo{
+        std::make_shared<Alcohol>(Alcohol("Whisky", 0, 100, 47)),
+        std::make_shared<Alcohol>(Alcohol("Beer", 0, 12, 8)),
+        std::make_shared<Alcohol>(Alcohol("Wine", 0, 40, 47)),
+        std::make_shared<Fruit>(Fruit("Banana", 0, 5, 10)),
+        std::make_shared<Fruit>(Fruit("Orange", 0, 7, 7)),
+        std::make_shared<Fruit>(Fruit("Raspberry", 0, 30, 2)),
+        std::make_shared<Fruit>(Fruit("GoldenCucumber", 0, 70, 1)),
+    };
+
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> initialAmount(0, 20);
+
+    for (const auto& cargo : possibleRegularCargo) {
+        *cargo += initialAmount(gen);
+        stock_.push_back(cargo);
+    }
+}
+
+void Store::generateItems() {
+    std::vector<std::shared_ptr<Cargo>> items{
+        std::make_shared<Item>(Item("MetalScrap", 10, 2, Item::Rarity::common)),
+        std::make_shared<Item>(Item("Sextant", 3, 120, Item::Rarity::rare)),
+        std::make_shared<Item>(Item("BlackPearl", 1, 400, Item::Rarity::epic)),
+        std::make_shared<Item>(Item("Kraken's eye", 1, 1000, Item::Rarity::legendary))
+
+    };
+
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> itemRarity(0, 100);
+
+    auto result = itemRarity(gen);
+
+    if (result > 50) {
+        stock_.push_back(items[0]);
+    }
+    if (result > 80) {
+        stock_.push_back(items[1]);
+    }
+    if (result > 95) {
+        stock_.push_back(items[2]);
+    }
+    if (result > 98) {
+        stock_.push_back(items[3]);
+    }
+}
```

Comment:
I would also put this inside a f-ction that takes one argument - items

Reply:
Done.